syntax/Jenkinsfile.vim

```diff
@@ -1,21 +1,22 @@
 runtime syntax/groovy.vim
 syn keyword jenkinsfileBuiltInVariable currentBuild
 
-syn keyword jenkinsfileSection pipeline agent stages steps post
+syn keyword jenkinsfileSection pipeline agent matrix axes excludes stages post
 
-syn keyword jenkinsfileDirective environment options parameters triggers stage tools input when libraries
+syn keyword jenkinsfileDirective environment options triggers tools input when libraries exclude
 
 syn keyword jenkinsfileOption contained buildDiscarder disableConcurrentBuilds overrideIndexTriggers skipDefaultCheckout nextgroup=jenkinsfileOptionParams
 syn keyword jenkinsfileOption contained skipStagesAfterUnstable checkoutToSubdirectory timeout retry timestamps nextgroup=jenkinsfileOptionParams
 syn keyword jenkinsfileOption contained disableResume newContainerPerStage preserveStashes quietPeriod parallelsAlwaysFailFast nextgroup=jenkinsfileOptionParams
 syn region  jenkinsfileOptionParams contained start='(' end=')' transparent contains=@groovyTop
 syn match   jenkinsfileOptionO /[a-zA-Z]\+([^)]*)/ contains=jenkinsfileOption,jenkinsfileOptionParams transparent containedin=groovyParenT1
 
+syn keyword jenkinsfileCoreStep axis skipwhite nextgroup=jenkinsfileAxisConfigBlock
 syn keyword jenkinsfileCoreStep checkout
-syn keyword jenkinsfileCoreStep docker dockerfile skipwhite nextgroup=jenkinsFileDockerConfigBlock
+syn keyword jenkinsfileCoreStep docker dockerfile skipwhite nextgroup=jenkinsfileDockerConfigBlock
 syn keyword jenkinsfileCoreStep node
+syn keyword jenkinsfileCoreStep parameters skipwhite nextgroup=jenkinsfileParametersConfigBlock
 syn keyword jenkinsfileCoreStep scm
-syn keyword jenkinsfileCoreStep sh
```

Comment:
The `sh` step is already a `jenkinsfilePipelineStep` below.  It's better suited there so this conflicting definition is removed.